service/controller/appvalue/watch.go

```diff
@@ -0,0 +1,138 @@
+package appvalue
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/giantswarm/app-operator/v2/pkg/annotation"
+)
+
+func (c *AppValue) watch(ctx context.Context) {
+	var lastResourceVersion string
+
+	for {
+		lo := metav1.ListOptions{
+			LabelSelector: c.selector.String(),
+		}
+
+		// Found the highest resourceVersion in cofigMaps CRs
+		cms, err := c.k8sClient.K8sClient().CoreV1().ConfigMaps("").List(ctx, lo)
+		if err != nil {
+			panic(err)
+		}
+
+		for _, cm := range cms.Items {
+			if lastResourceVersion < cm.ResourceVersion {
+				lastResourceVersion = cm.ResourceVersion
+			}
+		}
+
+		c.logger.Log("debug", fmt.Sprintf("starting ResourceVersion is %s", lastResourceVersion))
```

Comment:
Getting the highest resourceVersion among configMap objects and get updates event only if the resourceVersion is higher. 
This is not to trigger the app CRs with older events.